### PR TITLE
[Core/ContextMenu] ContextMenu respects dark theme

### LINKED ISSUE #776: Context Menu in Dark Theme
'Content Menu' in 'Dark theme' is White however popups are dark. Check below images.
Is this expected behaviour or a bug?  I was expecting Context menu to be dark in dark theme like popovers.

Same can be reproduced by going to docs page
http://blueprintjs.com/docs/#components.popover
and change theme to dark from top

screen shot 2017-03-04 at 3 42 03 pm

screen shot 2017-03-04 at 3 42 19 pm

### PR DESCRIPTION
**Pull request by mcintyret**
#### Changes proposed in this pull request:

Context menu respects dark theme.

#### Reviewers should focus on:

Is there a better way of doing this??

#### Screenshot

image
but still
image

### CODE REVIEW

File: packages/core/src/common/utils/isDarkTheme.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/*
```
Comment (cmslewis) [inline packages/core/src/common/utils/isDarkTheme.ts]: Let's rename this file to `themeUtils.ts` (bags of functions are always called `___Utils.ts` in our codebase).

File: packages/core/src/components/context-menu/contextMenuTarget.tsx [hunk 2]
```diff
@@ -6,9 +6,11 @@
  */
 
 import * as React from "react";
+import * as ReactDOM from "react-dom";
 
 import { CONTEXTMENU_WARN_DECORATOR_NO_METHOD } from "../../common/errors";
 import { isFunction, safeInvoke } from "../../common/utils";
+import { isDarkTheme } from "../../common/utils/isDarkTheme";
```
Comment (cmslewis) [inline packages/core/src/components/context-menu/contextMenuTarget.tsx]: Then change this to `import * as ThemeUtils from "../../common/utils/themeUtils"`

And reference the function as `ThemeUtils.isDarkTheme(...)`.

File: packages/core/src/components/context-menu/contextMenuTarget.tsx [hunk 3]
```diff
@@ -33,6 +35,8 @@ export function ContextMenuTarget<T extends { prototype: IContextMenuTarget }>(c
             // always return `element` in case caller is distinguishing between `null` and `undefined`
             return element;
         }
+        /* tslint:disable:no-this-assignment */
+        const thisElem = this;
```
Comment (cmslewis) [inline packages/core/src/components/context-menu/contextMenuTarget.tsx]: Doesn't the fat-arrow definition on `onContextMenu` maintain the same `this` binding as this parent scope? Seems like you don't need this.

File: packages/core/src/components/context-menu/contextMenuTarget.tsx [hunk 4]
```diff
@@ -44,8 +48,10 @@ export function ContextMenuTarget<T extends { prototype: IContextMenuTarget }>(c
             if (isFunction(this.renderContextMenu)) {
                 const menu = this.renderContextMenu(e);
                 if (menu != null) {
+                    const htmlElement = ReactDOM.findDOMNode(thisElem);
+                    const darkTheme = htmlElement && isDarkTheme(htmlElement);
```
Comment (cmslewis) [inline packages/core/src/components/context-menu/contextMenuTarget.tsx]: Use explicit `htmlElement != null`

File: packages/core/src/components/popover/popover.tsx [hunk 5]
```diff
@@ -18,6 +18,7 @@ import * as PosUtils from "../../common/position";
 import { IProps } from "../../common/props";
 import * as TetherUtils from "../../common/tetherUtils";
 import * as Utils from "../../common/utils";
+import { isDarkTheme } from "../../common/utils/isDarkTheme";
```
Comment (cmslewis) [inline packages/core/src/components/popover/popover.tsx]: `import * as ThemeUtils from...`

File: packages/core/src/common/utils/isDarkTheme.ts (hunk 1 above)
Comment (giladgray) [reply packages/core/src/common/utils/isDarkTheme.ts]: disagree! I've come to greatly prefer the pattern where you have a `utils/` _directory_ and a bunch of individual files, just like this.

File: packages/core/src/components/context-menu/contextMenuTarget.tsx (hunk 2 above)
Comment (giladgray) [reply packages/core/src/components/context-menu/contextMenuTarget.tsx]: disagree, let's stick with the individual files in `utils/` directory

File: packages/core/src/components/context-menu/contextMenuTarget.tsx (hunk 3 above)
Comment (giladgray) [reply packages/core/src/components/context-menu/contextMenuTarget.tsx]: yeah I don't think this is necessary at all.

@mcintyret please try removing this line and just using `this` directly below.

File: packages/core/src/common/utils/isDarkTheme.ts (hunk 1 above)
Comment (cmslewis) [reply packages/core/src/common/utils/isDarkTheme.ts]: I don't particularly want to have two schemes in play at the same time in the codebase, but okay.

**Pull request merged**